pvlib/soiling.py

```diff
@@ -69,15 +69,22 @@ def hsu(rainfall, cleaning_threshold, tilt, pm2_5, pm10,
 
     # never use mutable input arguments
     if depo_veloc is None:
-        depo_veloc = {'2_5': 0.004, '10': 0.0009}
+        depo_veloc = {'2_5': 0.0009, '10': 0.004}
 
     # accumulate rainfall into periods for comparison with threshold
     accum_rain = rainfall.rolling(rain_accum_period, closed='right').sum()
     # cleaning is True for intervals with rainfall greater than threshold
     cleaning_times = accum_rain.index[accum_rain >= cleaning_threshold]
+    
+    # determine the time intervals in seconds (dt_sec)
+    dt = rainfall.index.values       # datetimes in nanoseconds
+    dt2 = np.roll(dt.copy(),1)       # shift array values by one place
+    dt3 = np.delete((dt-dt2)/1e9,0)  # subtract shifted values from original and convert to seconds
+    # append the last value to the end so there are the same number of elements in the array
+    dt_sec = np.append(dt3,dt3[-1]).astype('float64') 
```

Comment:
This index -> interval length calculation was done in the snow module as well: https://github.com/pvlib/pvlib-python/blob/master/pvlib/snow.py#L11-L13

I wonder if that function should be moved to `pvlib.tools` and used here.

Reply:
I think you are referring to the function: _time_delta_in_hours().  It is similar to what I was doing, but it gives a NaN in the first position of the result.   I think we should leave it as it is and raise a new issue if needed.